### PR TITLE
🧹 Follow-up to fix some issues in the multi asset progressbars

### PR DESCRIPTION
**Pull request by czunker**
Signed-off-by: Christian Zunker <christian@mondoo.com>

### CODE REVIEW

**Comment by czunker** [comment]
This is a follow-up to https://github.com/mondoohq/cnquery/pull/771 and addresses the latest comments.

**Comment by czunker** [comment]
With the changed behavior of the overall progress bar, I also changed the text for it:

before:
100% overall 11/16 assets 5/16 errors

after:
100% overall 11/16 scanned 5/16 errored

At least to me this is more clear. Please double check.

File: cli/progress/multiprogress.go [hunk 1]
```diff
@@ -174,6 +175,7 @@ func (m modelMultiProgress) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 			m.Progress[msg.Index].lock.Unlock()
 		}
 
+		overallPercent := 0.0
 		if _, ok := m.Progress[overallProgressIndexName]; ok {
 			m.Progress[msg.Index].lock.Lock()
```
Comment (chris-rock) [inline cli/progress/multiprogress.go]: why is the lock on the index and not the overallProgressIndexName

**Comment by chris-rock** [comment]
On current branch this is where the progress bar is stuck. I had to ctrl+c to close the progress bar. Then the report rendered. As you can see the progress was not getting to 100. Is that caused by the fact that errored assets are not counted into the overall progress?

Screenshot 2023-01-23 at 14 20 10

File: cli/progress/multiprogress.go (hunk 1 above)
Comment (czunker) [reply cli/progress/multiprogress.go]: Good catch. Thanks. That was a Copy&Paste error.

**Comment by czunker** [comment]
> On current branch this is where the progress bar is stuck. I had to ctrl+c to close the progress bar. Then the report rendered. As you can see the progress was not getting to 100. Is that caused by the fact that errored assets are not counted into the overall progress?
> 
> Screenshot 2023-01-23 at 14 20 10

I changed the code to quit the tea program from inside the multi progress bar. That created some edge cases, because there are different situations when the program should end depending on whether it is used in cnquery or cnspec.

So, now the tea program is again closed when the scan ends. This will not work as soon as we want to run concurrent scans, but I'd like to change this in a separate PR. Giving it a quick try ended the program but left the CLI in a broken state.

**Review by chris-rock — APPROVED** [review]
Thank you @czunker

**Pull request merged**